src/hooks/useMagicNetwork.ts

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useCallback } from 'react'
+import { useSearchParams } from 'next/navigation'
+import { useAppDispatch } from '@/store'
+import { setRpc } from '@/store/settingsSlice'
+import { addChain } from '@/store/chainsSlice'
+import { type ChainInfo, type RPC_AUTHENTICATION } from '@safe-global/safe-gateway-typescript-sdk'
+import useChainId from '@/hooks/useChainId'
+import useChains from './useChains'
+
+const CHAINS_STORAGE_KEY = 'chains'
+
+export const useMagicNetwork = (): void => {
+  const searchParams = useSearchParams()
+  const dispatch = useAppDispatch()
+  const chainId = useChainId()
+  const supportedChains = useChains()
+
+  useEffect(() => {
+    // Get params
+    const chainIdParam = searchParams.get('chainId')
+    const rpcUrl = searchParams.get('rpc')
+    const shortName = searchParams.get('chain')
+    const currencyName = searchParams.get('currency')
+    const currencySymbol = searchParams.get('symbol')
+    const currencyLogo = searchParams.get('logo')
+    const explorerAddr = searchParams.get('expAddr')
+    const explorerTx = searchParams.get('expTx')
+
+    // Return if no RPC param or chainId
+    if (!rpcUrl || !chainIdParam) return
+
+    // Check if chain already exists in supported chains
+    const existingChain = supportedChains.configs.find((chain) => chain.chainId === chainIdParam)
+
+    if (!existingChain) {
+      // Return if no currency info
+      if (!currencyName || !currencySymbol || !shortName) return
```

Comment:
I think we should show an error here saying that some required information was missing. You can see an example of an error notification here: https://github.com/eternalsafe/wallet/blob/eternalsafe/src/hooks/useMagicLink.ts#L64